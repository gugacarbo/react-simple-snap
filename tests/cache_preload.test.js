const fs = require("fs");
const writeFileSpy = jest.spyOn(fs, "writeFile");
writeFileSpy.mockImplementation((file, data, cb) => cb());

const { mockFs } = require("./helper.js");
const snapRun = require("./snapRun");

describe("cacheAjaxRequests", () => {
  const source = "tests/examples/other";
  const include = ["/ajax-request.html"];
  const { fs, filesCreated, content } = mockFs();
  beforeAll(() => snapRun(fs, { source, include, cacheAjaxRequests: true }));
  test("saves ajax response", () => {
    expect(filesCreated()).toEqual(1);
    expect(content(0)).toMatch(
      'window.snapStore={"\\u002Fjs\\u002Ftest.json":{"test":1}};'
    );
  });
});

describe("preconnectThirdParty", () => {
  const source = "tests/examples/other";
  const include = ["/third-party-resource.html"];
  const { fs, filesCreated, content } = mockFs();
  beforeAll(() => snapRun(fs, { source, include }));
  test("adds <link rel=preconnect>", () => {
    expect(filesCreated()).toEqual(1);
    expect(content(0)).toMatch(
      '<link href="https://fonts.googleapis.com" rel="preconnect">'
    );
  });
});

describe("http2PushManifest", () => {
  const source = "tests/examples/other";
  const include = ["/with-big-css.html"];
  const { fs, filesCreated, content } = mockFs();
  beforeAll(() => snapRun(fs, { source, include, http2PushManifest: true }));
  test("writes http2 manifest file", () => {
    expect(filesCreated()).toEqual(2);
    expect(content(1)).toEqual(
      '[{"source":"/with-big-css.html","headers":[{"key":"Link","value":"</css/big.css>;rel=preload;as=style"}]}]'
    );
  });
});

describe("ignoreForPreload", () => {
  const source = "tests/examples/other";
  const include = ["/with-big-css.html"];
  const { fs, filesCreated, content } = mockFs();
  beforeAll(() =>
    snapRun(fs, {
      source,
      include,
      http2PushManifest: true,
      ignoreForPreload: ["big.css"],
    })
  );
  test("writes http2 manifest file", () => {
    expect(filesCreated()).toEqual(2);
    expect(content(1)).toEqual("[]");
  });
});

describe("preloadImages", () => {
  const source = "tests/examples/other";
  const include = ["/with-image.html"];
  const { fs, filesCreated, content } = mockFs();
  beforeAll(() => snapRun(fs, { source, include, preloadImages: true }));
  test("adds <link rel=preconnect>", () => {
    expect(filesCreated()).toEqual(1);
    expect(content(0)).toMatch(
      '<link as="image" href="/css/bg.png" rel="preload">'
    );
  });
});
