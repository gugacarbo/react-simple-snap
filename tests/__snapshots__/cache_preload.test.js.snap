// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sitemap sitemap 1`] = `""<?xml version=\"1.0\" encoding=\"UTF-8\"?><urlset xmlns=\"http://www.sitemaps.org/schemas/sitemap/0.9\" xmlns:news=\"http://www.google.com/schemas/sitemap-news/0.9\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\" xmlns:image=\"http://www.google.com/schemas/sitemap-image/1.1\" xmlns:video=\"http://www.google.com/schemas/sitemap-video/1.1\"><url><loc>https://www.yourprerendereddomain.com/</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/5</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/404.html</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/1</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/2</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/4</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/3</loc><changefreq>weekly</changefreq><priority>1.0</priority></url></urlset>""                                  
    + ""<?xml version=\"1.0\" encoding=\"UTF-8\"?><urlset xmlns=\"http://www.sitemaps.org/schemas/sitemap/0.9\" xmlns:news=\"http://www.google.com/schemas/sitemap-news/0.9\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\" xmlns:image=\"http://www.google.com/schemas/sitemap-image/1.1\" xmlns:video=\"http://www.google.com/schemas/sitemap-video/1.1\"><url><loc>https://www.yourprerendereddomain.com/</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/404.html</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/5</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/1</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/2</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/3</loc><changefreq>weekly</changefreq><priority>1.0</priority></url><url><loc>https://www.yourprerendereddomain.com/4</loc><changefreq>weekly</changefreq><priority>1.0</priority></url></urlset>""`;
